app/code/Magento/InventoryGraphQl/Model/Resolver/OnlyXLeftInStockResolver.php

```diff
@@ -0,0 +1,107 @@
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\InventoryCatalog\Model\GetStockIdForCurrentWebsite;
+use Magento\InventoryConfigurationApi\Api\GetStockItemConfigurationInterface;
+use Magento\InventoryConfigurationApi\Exception\SkuIsNotAssignedToStockException;
+use Magento\InventorySalesApi\Api\GetProductSalableQtyInterface;
+
+/**
+ * @inheritdoc
+ */
+class OnlyXLeftInStockResolver implements ResolverInterface
+{
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var GetProductSalableQtyInterface
+     */
+    private $getProductSalableQty;
+
+    /**
+     * @var GetStockIdForCurrentWebsite
+     */
+    private $getStockIdForCurrentWebsite;
+
+    /**
+     * @var GetStockItemConfigurationInterface
+     */
+    private $getStockItemConfiguration;
+
+    /**
+     * @param ScopeConfigInterface $scopeConfig
+     * @param GetProductSalableQtyInterface $getProductSalableQty
+     * @param GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite
+     * @param GetStockItemConfigurationInterface $getStockItemConfiguration
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        GetProductSalableQtyInterface $getProductSalableQty,
+        GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite,
+        GetStockItemConfigurationInterface $getStockItemConfiguration
+    ) {
+        $this->scopeConfig = $scopeConfig;
+        $this->getProductSalableQty = $getProductSalableQty;
+        $this->getStockIdForCurrentWebsite = $getStockIdForCurrentWebsite;
+        $this->getStockItemConfiguration = $getStockItemConfiguration;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
+    {
+        /* @var $product ProductInterface */
+        $product = $value['model'];
+        $onlyXLeftQty = $this->getOnlyXLeftQty($product->getSku());
+
+        return $onlyXLeftQty;
+    }
+
+    /**
+     * @param string $sku
+     *
+     * @return null|float
+     * @throws SkuIsNotAssignedToStockException
+     * @throws LocalizedException
+     */
+    private function getOnlyXLeftQty(string $sku): ?float
+    {
+        $stockId = $this->getStockIdForCurrentWebsite->execute();
+        $stockItemConfiguration = $this->getStockItemConfiguration->execute($sku, $stockId);
+
+        $thresholdQty = $stockItemConfiguration->getStockThresholdQty();
+        if ($thresholdQty === 0) {
+            return null;
+        }
+
+        try {
+            $productSalableQty = $this->getProductSalableQty->execute($sku, $stockId);
+            $stockLeft = $productSalableQty - $stockItemConfiguration->getMinQty();
+
+            if ($productSalableQty > 0 && $stockLeft <= $thresholdQty) {
+                return (float)$stockLeft;
+            }
+        } catch (InputException | LocalizedException $e) {
+            return null;
```

Comment:
Should we provide fallback to `Magento\CatalogInventoryGraphQl\Model\Resolver\OnlyXLeftInStockResolver` when we have exception like "Can't check requested quantity for products without Source Items support" ?